src/battle_script_commands.c

```diff
@@ -11990,6 +11990,11 @@ static void Cmd_handleballthrow(void)
             }
             else // not caught
             {
+                if (!gHasFetchedBall)
+                {
+                    gLastUsedBall = gLastUsedItem;
+                    gBattleResults.catchAttempts[gLastUsedBall]++;
```

Comment:
The Master Ball doesn't need to be taken into account.

It won't fail in a wild battle, and if lost by misusing it on a trainer battle, a Pokémon with Ball Fetch does not get it back, just like in Sword and Shield :eyes:

Reply:
Also, that check is not redundant because the one at L11932 affects gLastUsedItem, not gLastUsedBall.

I did try to make use of that originally, but I couldn't. That is the reason why I had to add gLastUsedBall :eyes: